src/graphics/vulkan/vulkan_asset_context.cpp

```
@@ -0,0 +1,44 @@
+#include "canyon/graphics/vulkan/vulkan_asset_context.h"
+#include "canyon/graphics/vulkan/vulkan_surface_context.h"
+#include "canyon/graphics/vulkan/vulkan_font.h"
+#include "canyon/graphics/vulkan/vulkan_image.h"
+#include "canyon/graphics/vulkan/vulkan_texture.h"
+
+namespace canyon::graphics::vulkan {
+    AssetContext::AssetContext(SurfaceContext& context)
+        : m_context(context) {
+    }
+
+    std::unique_ptr<IImage> AssetContext::NewImage(std::shared_ptr<ITexture> texture) {
```

Clang (14.0.6)

[error] variable 'NewImage' is non-const and globally accessible, consider making it const

(cppcoreguidelines-avoid-non-const-global-variables,-warnings-as-errors)